utilities/test_suite/HOST/Tensor_misc_host.cpp

```diff
@@ -0,0 +1,459 @@
+/*
+Copyright (c) 2019 - 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <iostream>
+#include <time.h>
+#include <omp.h>
+#include <fstream>
+#include <unistd.h>
+#include <dirent.h>
+#include <cstdlib>
+#include <ctime>
+#include <algorithm>
+#include <iterator>
+#include "rpp.h"
+#include "../rpp_test_suite_common.h"
+
+using namespace std;
+
+std::map<int, string> augmentationMiscMap =
+{
+    {0, "transpose"}
+};
+
+void compute_strides(RpptGenericDescPtr descriptorPtr)
+{
+    if (descriptorPtr->numDims > 0)
+    {
+        uint64_t v = 1;
+        for (int i = descriptorPtr->numDims - 1; i > 0; i--)
+        {
+            descriptorPtr->strides[i] = v;
+            v *= descriptorPtr->dims[i];
+        }
+        descriptorPtr->strides[0] = v;
+    }
+}
+
+string get_path(Rpp32u nDim, Rpp32u readType)
+{
+    string refPath = get_current_dir_name();
+    string pattern = "HOST/build";
```

Comment:
Tensor_misc_hip is mostly using same functions here. Suggest creating a header for these and we can include it in the HOST and HIP versions. 
Backend type can be passed into this function, so will work with - string pattern = "HOST/build" or "HIP/build"